### DISCUSSION TITLE
Failing to get mathlib4

### DISCUSSION THREAD

**Discussion by AndreSlavescu** [lean-dojo/ReProver#48] [Q&A] [answered]
Hello, 

I am trying to run the retriever example with the following:

python retrieval/main.py predict --config retrieval/confs/cli_lean4_random.yaml --ckpt_path "my path"

I saved the model checkpoints locally for the following model:

[kaiyuy/leandojo-lean4-retriever-byt5-small](https://huggingface.co/kaiyuy/leandojo-lean4-retriever-byt5-small)

And was able to get passed the _load_data stage where the examples are being loaded. After that stage where I get the following error:

Request GET /repos/leanprover-community/mathlib4 failed with 403: rate limit exceeded
Setting next backoff to 970.904135s

**Comment by yangky11** [comment, marked as answer]
Did you try set the `GITHUB_ACCESS_TOKEN` environment variable as mentioned in [LeanDojo's requirements](https://github.com/lean-dojo/LeanDojo?tab=readme-ov-file#requirements)?
